docs/reference/projectyml.rst

```diff
@@ -50,66 +50,165 @@ The code snippet below provides an example of a complete ``project.yml`` file:
+option e.g.:
 
+.. code-block:: yaml
 
-Alternatively, a pip config file can be used at a custom location.
-The ``index-url`` can be specified in this file as explained in the previous section.
-To make this work, the ``PIP_CONFIG_FILE`` environment variable needs to be set to the path of the newly created file (See: :ref:`env_vars`).
-For more information see the `Pip documentation <https://pip.pypa.io/en/stable/topics/configuration/>`_.
+    pip:
+        extra-index-url: [https://my_repo.secure.company.com/repository/project-alpha-production]
+        pre: false
+        use-system-config: true
 
-Specify the index-urls in the project.yml file
-""""""""""""""""""""""""""""""""""""""""""""""
 
-Another option is to use the  ``index_urls`` option in the ``pip`` section of the ``project.yml`` file:
+On the other hand, if the pip configuration is not fully managed at the system level, make sure to disable ``use-system-config``
+and set ``index-url`` to the secure internal repo e.g.:
 
 .. code-block:: yaml
 
     pip:
-      use_config_file: False
-      index_urls:
-        - <url of a python package repository>
+        index-url: https://my_repo.secure.company.com/repository/inmanta-production
+        extra-index-url: [https://my_repo.secure.company.com/repository/project-alpha-production]
+        pre: false
+        use-system-config: false
+
+.. note::
+    The ``pre`` and ``index-url`` options defined in the ``project.yml`` pip section will always take precedence over
+    the corresponding pip options, even when ``use-system-config`` is set to true.
+
+    The ``extra-index-url`` option will extend the extra indexes already set at the system level (if any).
+
+    Other pip-related environment variables are not overridden.
+
+    To use the system's pip configuration without altering it, leave the corresponding option unset.
 
 
 .. note::
-    The pip config file can also be used in combination with ``index-urls`` specified in the ``pip`` section of the ``project.yml`` file:
 
-    * If the pip config is used (by setting ``use_config_file`` to ``true``), the ``index-url`` specified in the pip config file will take precedence and the ``index-urls`` specified in the ``pip`` section of the ``project.yml`` file will be used as ``extra-index-urls`` when installing with pip.
-    * If the pip config is not used (by setting ``use_config_file`` to ``False``), then the first ``index_url`` specified in the project.yml will be used as an ``index_url`` and all the following ones will be used as ``extra-index-urls`` when installing with pip.
+    Set up authentication towards the index using netrc. See this :ref:`section<setting_up_pip_index_authentication>`
+    for more information.
+
+.. _migrate_to_project_wide_pip_config:
+
+Migrate to project-wide pip config
+----------------------------------
+
+``inmanta-core 11.0.0`` introduced new options to configure pip settings for the whole project in a centralized way.
```

Comment:
I would add a code sample here (or even add a script to the orchestrator to update all configs) 

This should be very copy-pasteable.

Reply:
Added a code sample